src/test/java/com/es/phoneshop/model/product/ArrayListProductDaoTest.java

```diff
@@ -15,7 +17,40 @@ public void setup() {
     }
 
     @Test
-    public void testFindProductsNoResults() {
-        assertTrue(productDao.findProducts().isEmpty());
+    public void deleteProductTest() {
+        Product testProduct = new Product(23L, "simsxg75", "Siemens SXG75", new BigDecimal(150), Currency.getInstance("USD"), 40,
+                "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Siemens/Siemens%20SXG75.jpg");
+        productDao.delete(23L);
+        assertEquals(productDao.getProduct(23L), testProduct);
+    }
+
+    @Test
+    public void getProductTest() {
+        Product testProduct = new Product(33L, "simsxg75", "Siemens SXG75", new BigDecimal(150), Currency.getInstance("USD"), 40,
+                "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Siemens/Siemens%20SXG75.jpg");
+        productDao.save(testProduct);
+        assertEquals(productDao.getProduct(33L), testProduct);
+    }
+
+    @Test
+    public void saveProductTest() {
+        Product testProduct = new Product(43L, "simsxg75", "Siemens SXG75", new BigDecimal(150), Currency.getInstance("USD"), 40,
+                "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Siemens/Siemens%20SXG75.jpg");
+        productDao.save(testProduct);
+        assertEquals(productDao.getProduct(43L), testProduct);
+    }
+
+
+    @Test
+    public void findProductsTest() {
+        Product testProductWithNullPrice = new Product(53L, "simsxg75", "Siemens SXG75", null, Currency.getInstance("USD"), 40,
+                "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Siemens/Siemens%20SXG75.jpg");
+        productDao.save(testProductWithNullPrice);
+        assertFalse(productDao.findProducts().contains(testProductWithNullPrice));
```

Comment:
what's wrong?

Reply:
BigDecimal nullPrise = null;
        Product testProductWithNullPrice = new Product(53L, "simsxg75", "Siemens SXG75", nullPrise, Currency.getInstance("USD"), 40,
                "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Siemens/Siemens%20SXG75.jpg");

Reply:
Hint: in real application in your case the assert statement would always be `false`. That means your test would never fail regardless the products you have in the storage. Why?